### ISSUE TITLE
SQL Server: Key is not the same length or scale as referencing column

### ISSUE THREAD

**Issue by scottbrady91** [dotnet/efcore#6449]
#### DbContext

```
public class ClientDbContext : DbContext
{
    public ClientDbContext(DbContextOptions options) : base(options) { }

    public DbSet<Client> Clients { get; set; }

    protected override void OnModelCreating(ModelBuilder modelBuilder)
    {
        var client = modelBuilder.Entity<Client>();
        client.ToTable("Clients");
        client.HasKey(x => x.ClientId);
        client.Property(x => x.ClientId).HasMaxLength(200);
        client.HasMany(x => x.RedirectUris).WithOne();

        base.OnModelCreating(modelBuilder);
    }
}
```
#### Entities

```
public class Client
{
    public string ClientId { get; set; }
    public List<ClientRedirectUri> RedirectUris { get; set; }
}

public class ClientRedirectUri
{
    public int Id { get; set; }
    public string RedirectUri { get; set; }
}
```
### The issue

Trying to use a context with a max length on its primary key, getting hit by an exception previously reported in issues #275, #2789, #2455 and #6236. Haven't seen issue reported for 1.0.0.

Issue persists if I explicitly state foreign key relationships, but works if I duplicate the max length logic on the foreign key.

Logic works for SQLite and InMemory Database Providers, only experiences issue with SQL Server.

```
Exception message: Column 'Clients.ClientId' is not the same length or scale as referencing column 'ClientRedirectUri.ClientId' in foreign key 'FK_ClientRedirectUri_Clients_ClientId'. Columns participating in a foreign key relationship must be defined with the same length and scale.
Could not create constraint or index. See previous errors.
Stack trace:    at System.Data.SqlClient.SqlConnection.OnError(SqlException exception, Boolean breakConnection, Action`1 wrapCloseInAction)
   at System.Data.SqlClient.TdsParser.ThrowExceptionAndWarning(TdsParserStateObject stateObj, Boolean callerHasConnectionLock, Boolean asyncClose)
   at System.Data.SqlClient.TdsParser.TryRun(RunBehavior runBehavior, SqlCommand cmdHandler, SqlDataReader dataStream, BulkCopySimpleResultSet bulkCopyHandler, TdsParserStateObject stateObj, Boolean& dataReady)
   at System.Data.SqlClient.SqlCommand.RunExecuteNonQueryTds(String methodName, Boolean async, Int32 timeout, Boolean asyncWrite)
   at System.Data.SqlClient.SqlCommand.InternalExecuteNonQuery(TaskCompletionSource`1 completion, Boolean sendToPipe, Int32 timeout, Boolean asyncWrite, String methodName)
   at System.Data.SqlClient.SqlCommand.ExecuteNonQuery()
   at Microsoft.EntityFrameworkCore.Storage.Internal.RelationalCommand.Execute(IRelationalConnection connection, String executeMethod, IReadOnlyDictionary`2 parameterValues, Boolean openConnection, Boolean closeConnection)
   at Microsoft.EntityFrameworkCore.Storage.Internal.RelationalCommand.ExecuteNonQuery(IRelationalConnection connection, IReadOnlyDictionary`2 parameterValues, Boolean manageConnection)
   at Microsoft.EntityFrameworkCore.Migrations.Internal.MigrationCommandExecutor.ExecuteNonQuery(IEnumerable`1 migrationCommands, IRelationalConnection connection)
   at Microsoft.EntityFrameworkCore.Storage.RelationalDatabaseCreator.EnsureCreated()
   at ConsoleApp1.Program.Main(String[] args) in E:\scratch\EfCoreRelationalTest\ConsoleApp1\Program.cs:line 21
```
### Further technical details

EF Core version: 1.0.0
Operating system: Windows 10 64-bit
Visual Studio version: VS 2015

**Comment by smitpatel** [comment]
Dupe of #2455 It has been fixed in 1.1.0 and will be available in next release.

**Comment by scottbrady91** [comment]
Cool! Any idea when?
